src/Metadata/Operation.php

```diff
@@ -90,8 +92,15 @@ public function __construct(
         protected ?string $name = null,
         $provider = null,
         $processor = null,
-        protected array $extraProperties = []
+        protected array $extraProperties = [],
+        protected ?PersistenceMeansInterface $persistenceMeans = null,
     ) {
+        if (null !== $this->elasticsearch) {
+            $solution = $this->elasticsearch
+                ? sprintf('Pass an instance of %s to $persistenceMeans instead', ElasticsearchDocument::class)
+                : 'You will have to remove it when upgrading to v4';
+            trigger_deprecation('api-platform/core', '3.1', sprintf('Setting "elasticsearch" is deprecated. %s', $solution));
```

Comment:
Please avoid deprecating here and prefer to put this logic only in the deprecated factory.

Reply:
Deprecating a method parameter seems logical to do in the same method. Why are you suggesting otherwise?

Reply:
tldr: move the deprecation to ElasticsearchProviderResourceMetadataCollectionFactory